src/kube-runtime/src/plugins/teamwise_storage/init.py

```
@@ -15,167 +15,36 @@
-            command for mount_command in second_round_mount_commands for command in mount_command]
-
-        # 5 assemble all commands
-        mount_commands.extend(premount_commands + first_round_mount_commands +
-                              mkdir_commands + post_mount_commands + second_round_mount_commands)
-
-    return mount_commands
-
-
-def generate_plugin_commands(parameters) -> list:
-    resp = http_get("{}/api/v2/user/{}".format(REST_API_PREFIX, USER_NAME))
-    if resp.status_code != http.HTTPStatus.OK:
-        LOGGER.error("Failed to get user config, resp: %s", resp.text)
-        sys.exit(1)
-
-    user_config = resp.json()
-    if not user_config["storageConfig"]:
-        LOGGER.error(
-            "User %s don't has the permission to access storage", USER_NAME)
-        sys.exit(1)
-
-    user_storage_config = user_config["storageConfig"]
-
-    if not parameters or not parameters["storageConfigNames"]:
-        default_storage_configs = get_user_default_storage_config_names(user_storage_config)
-        if not default_storage_configs:
-            LOGGER.error("Not set default stroage config for user %s, please contect admin", USER_NAME)
-            sys.exit(1)
-        storage_config_names = default_storage_configs
-    else:
-        storage_config_names = parameters["storageConfigNames"]
-        if not StorageHelper.is_valid_storage_config(user_storage_config, storage_config_names):
-            LOGGER.error("User %s do not has permission to access storages: %s",
-                         USER_NAME, storage_config_names)
-            sys.exit(1)
-
-    return STORAGE_PRE_COMMAND + generate_commands(storage_config_names)
-
 
 def main():
     LOGGER.info("Preparing storage runtime plugin commands")
     [parameters, pre_script, _] = plugin_init()
-    pre_script_commands = generate_plugin_commands(parameters)
+
+    try:
+        command_generator = StorageCommandGenerator()
+    except Exception:  #pylint: disable=broad-except
```

specify exception types?

---

Catch all exception by purpose, want to return unified code for error recognition

---
In reply to: [352484223](https://github.com/microsoft/pai/pull/3944#discussion_r352484223) [](ancestors = 352484223)